File: solutions/elixir/150-design-hit-counter.ex
# Problem 150: Design Hit Counter (LeetCode 362)
# Difficulty: Med
# Language: Elixir
# 
defmodule HitCounter do
  def new_counter(), do: []
  def hit(counter, timestamp), do: counter ++ [timestamp]
  def get_hits(counter, timestamp) do
    filtered = Enum.filter(counter, &(&1 > timestamp - 300))
    {length(filtered), filtered}
  end
end

File: solutions/typescript/150-design-hit-counter.ts
/**
 * Problem 150: Design Hit Counter (LeetCode 362)
 * Difficulty: Med
 * Language: TypeScript
 */
class HitCounter {
  private hits: number[] = [];
  hit(timestamp: number): void { this.hits = [...this.hits, timestamp]; }
  getHits(timestamp: number): number {
    this.hits = this.hits.filter(t => t > timestamp - 300);
    return this.hits.length;
  }
}
